File: program_data/github_java_sort_function/2/43.java

    public static void sort(int[] A) {
        int[] temp = new int[A.length];
        int start = 0;
        int end = A.length - 1;
        mergeSort(A, start, end, temp);
    }

    private static void mergeSort(int[] A, int start, int end, int[] temp) {
        if (start >= end) {
            return;
        }

        int mid = start + (end - start) / 2;
        mergeSort(A, start, mid, temp);
        mergeSort(A, mid + 1, end, temp);
        merge(A, start, mid, end, temp);
    }

    private static void merge(int[] A, int start, int mid, int end, int[] temp) {

        int left = start;
        int right = mid + 1;

        
        int k = start;

        
        while (left <= mid && right <= end) {
            if (A[left] < A[right]) {
                temp[k++] = A[left++];
            } else {
                temp[k++] = A[right++];
            }
        }
        while (left <= mid) {
            temp[k++] = A[left++];
        }
        while (right <= end) {
            temp[k++] = A[right++];
        }

        
        for (k = start; k <= end; k++) {
            A[k] = temp[k];
        }
        
        
        
        
        
    }

File: program_data/PKU_raw/76/43.c
struct qujian
{
	int a;
	int b;
}qj[50000];

int main()
{
	int n,i,j,temp,left,right,k;
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		scanf("%d%d",&qj[i].a,&qj[i].b);
	}
	for(i=0;i<n;i++)
	{
		for(j=i+1;j<n;j++)
		{
			if(qj[i].a>qj[j].a)
			{
				temp=qj[i].a;
				qj[i].a=qj[j].a;
				qj[j].a=temp;
				temp=qj[i].b;
				qj[i].b=qj[j].b;
				qj[j].b=temp;
			}
		}
	}
	left=qj[0].a;
	right=qj[0].b;
	k=1;
	while(k>0)
	{
		if(k==n)
		{
			printf("%d %d\n",left,right);
			break;
		}
		else
		{
			if(qj[k].a>right)
			{
				printf("no\n");
				break;
			}
			else
			{
				if(qj[k].a<left)
				{
					left=qj[k].a;
				}
				if(qj[k].b>right)
				{
					right=qj[k].b;
				}
				k++;
			}
		}
	}
	return 0;
}



	
	